### DISCUSSION TITLE
Default size of window

### DISCUSSION THREAD

**Discussion by norricorp** [slint-ui/slint#4827] [Q&A]
Is there a list of default sizes for windows and other elements.
On my screen (QHD) if a window is created without width / height then it opens to about 500px square.
Am I right in thinking window does not have a parent? So I can't open a window that is half the width of the screen and .9 of the height?
regards,
John

**Comment by ogoffart** [comment]
You can use preferred-width/preferred-height to give a different default size.

> Am I right in thinking window does not have a parent?

Yes that's right

> So I can't open a window that is half the width of the screen and .9 of the height?

We don't have public API to access the size of the screen at this moment.
(It might be possible with private API by accessing winit API dirrectly)

**Comment by norricorp** [comment]
Hi Oliver, thanks for this.
Interestingly I added preferred-* and it made absolutely no difference. Same default size as having not set size at all.
my code is 

		export component Home inherits Window {

			preferred-height: 1200px;
			preferred-width: 1000px;

			VerticalLayout {
				Rectangle {
					background: blue; 
					HorizontalLayout {
						
					}
				}
			}
		}

Is this a bug?

Regards,
John

**Comment by norricorp** [comment]
And after I added a Button to the Rectangle, suddenly preferred-size worked.
